### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by ardalis**
Please review @rynowak

### CODE REVIEW

**Comment by ardalis** [comment]
Fixes #203

**Comment by ardalis** [comment]
@danroth27 ship it?

**Comment by danroth27** [comment]
@Tratcher Please review

File: aspnet/testing/unit-testing.rst [hunk 1]
```diff
@@ -2,28 +2,131 @@ Unit Testing
...
 
-Unit Testing
-------------
-About unit testing
+.. note:: A `unit test naming convention <http://ardalis.com/unit-test-naming-convention>`_ I've found to be very effective is to name the class and test such that it forms a sentence, e.g. ``PricingCalculatorShould.ReturnZeroGivenQuantityZero()``. This makes for very readable test results, and tends to be far more helpful than naming test classes "FooTests" or "BarTests".
```
Comment (Tratcher) [inline aspnet/testing/unit-testing.rst]: Here are our naming guidelines: https://github.com/aspnet/Home/wiki/Engineering-guidelines#unit-test-method-naming We shouldn't make recommendations in the docs beyond what we use ourselves.

The tone is also out of place for docs, there shouldn't be any personal "I" references.

File: aspnet/testing/unit-testing.rst [hunk 2]
```diff
@@ -2,28 +2,131 @@ Unit Testing
...
+A test project is just a class library with references to a test runner and the project being tested (also referred to as the System Under Test or SUT). It's a good idea to organize your test projects in a separate folder from your SUT projects, and the recommended convention for DNX projects is something like this:
+
+.. code-block:: text
+
+	global.json
+	PrimeWeb.sln
+	src/
+	   PrimeWeb/
+	      project.json
+		  Startup.cs
```
Comment (Tratcher) [inline aspnet/testing/unit-testing.rst]: the spacing looks off here. Startup.cs and project.json should be on the same level.

File: aspnet/testing/unit-testing.rst [hunk 3]
```diff
@@ -2,28 +2,131 @@ Unit Testing
...
+
+.. code-block:: powershell
+
+	dnx test
+
+You should see output similar to the following:
+
+.. image:: unit-testing/_static/dnx-test.png
+
+dnx-watch
```
Comment (Tratcher) [inline aspnet/testing/unit-testing.rst]: @victorhurdugaci Is this a recommended use of dnx-watch?

File: aspnet/testing/unit-testing.rst [hunk 4]
```diff
@@ -2,28 +2,131 @@ Unit Testing
...
+^^^^^^^^^
+You can use the ``dnx-watch`` command to automatically execute a dnx command whenever the contents of the folder change. This can be used to automatically run tests whenever files are saved in the project. Note that it will detect changes to both the SUT project and the test project, even when run from the test project folder.
+
+.. note:: Learn more about `getting started with dnx-watch <http://ardalis.com/get-started-with-dnx-watch>`_.
```
Comment (Tratcher) [inline aspnet/testing/unit-testing.rst]: @danroth27 External blog link?

File: aspnet/testing/unit-testing.rst (hunk 3 above)
Comment (victorhurdugaci) [inline aspnet/testing/unit-testing.rst]: Assuming that tests have a direct reference to the project being tested then it is a perfectly valid scenario

File: aspnet/testing/unit-testing/sample/src/PrimeWeb/Services/PrimeService.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class PrimeService
+    {
+        public bool IsPrime(int candidate)
+        {
+            if(candidate < 2)
+            {
+                return false;
+            }
+            for(int divisor=2;divisor <= Math.Sqrt(candidate);divisor++)
```
Comment (Tratcher) [inline aspnet/testing/unit-testing/sample/src/PrimeWeb/Services/PrimeService.cs]: Nit: more spacing

File: aspnet/testing/unit-testing/sample/src/PrimeWeb/Startup.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        // For more information on how to configure your application, visit http://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            // Add the platform handler to the request pipeline.
+            app.UseIISPlatformHandler();
+            app.UseDeveloperExceptionPage();
```
Comment (Tratcher) [inline aspnet/testing/unit-testing/sample/src/PrimeWeb/Startup.cs]: UseDeveloperExceptionPage should only be referenced in a dev environment. See the templates.

File: aspnet/testing/unit-testing/sample/test/PrimeWeb.UnitTests/project.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        "System.Collections": "4.0.11-beta-23409",
+        "System.Linq": "4.0.1-beta-23409",
+        "System.Runtime": "4.0.21-beta-23409",
+        "System.Threading": "4.0.11-beta-23409"
+      }
+    }
+  },
+  "dependencies": {
+    "PrimeWeb": "1.0.0-*",
+    "xunit": "2.1.0-*",
```
Comment (Tratcher) [inline aspnet/testing/unit-testing/sample/test/PrimeWeb.UnitTests/project.json]: Fixed version?

File: aspnet/testing/unit-testing.rst (hunk 1 above)
Comment (ardalis) [inline aspnet/testing/unit-testing.rst]: Updated the note to refer to the Engineering-guidelines.

File: aspnet/testing/unit-testing.rst (hunk 2 above)
Comment (ardalis) [inline aspnet/testing/unit-testing.rst]: Was a tabs vs. spaces issue. corrected.

File: aspnet/testing/unit-testing.rst (hunk 4 above)
Comment (ardalis) [inline aspnet/testing/unit-testing.rst]: Not the first time.

File: aspnet/testing/unit-testing/sample/src/PrimeWeb/Services/PrimeService.cs (hunk 5 above)
Comment (ardalis) [inline aspnet/testing/unit-testing/sample/src/PrimeWeb/Services/PrimeService.cs]: Added a couple.

File: aspnet/testing/unit-testing/sample/test/PrimeWeb.UnitTests/project.json (hunk 7 above)
Comment (ardalis) [inline aspnet/testing/unit-testing/sample/test/PrimeWeb.UnitTests/project.json]: Are you saying I shouldn't use it (lines 12-16) or I should (lines 21-22), or ?

Comment (Tratcher) [inline aspnet/testing/unit-testing/sample/test/PrimeWeb.UnitTests/project.json]: Samples targeting released versions (beta8) should use fixed package dependencies.

Comment (ardalis) [inline aspnet/testing/unit-testing/sample/test/PrimeWeb.UnitTests/project.json]: done

**Pull request merged**